zio-http/shared/src/main/scala/zio/http/codec/ContentCodecs.scala

```diff
@@ -16,40 +16,48 @@
 
 package zio.http.codec
 
-import zio.stacktracer.TracingImplicits.disableAutoTrace
-
 import zio.stream.ZStream
 
-import zio.schema.Schema
-
 import zio.http.MediaType
 
 private[codec] trait ContentCodecs {
-  def content[A](name: String)(implicit schema: Schema[A]): ContentCodec[A] =
-    HttpCodec.Content(schema, mediaType = None, Some(name))
+  def content[A](name: String)(implicit codec: HttpContentCodec[A]): ContentCodec[A] =
+    HttpCodec.Content(codec, Some(name))
 
-  def content[A](implicit schema: Schema[A]): ContentCodec[A] =
-    HttpCodec.Content(schema, mediaType = None, None)
+  def content[A](implicit codec: HttpContentCodec[A]): ContentCodec[A] =
+    HttpCodec.Content(codec, None)
 
-  def content[A](name: String, mediaType: MediaType)(implicit schema: Schema[A]): ContentCodec[A] =
-    HttpCodec.Content(schema, mediaType = Some(mediaType), Some(name))
+  def content[A](name: String, mediaType: MediaType)(implicit codec: HttpContentCodec[A]): ContentCodec[A] =
```

Comment:
This makes me want to add a covariant phantom type to `HttpCodecContent`, some `MediaTypes` (`<: MediaType`), such that this could be written:

```suggestion
  def content[A, M <: MediaType](name: String, mediaType: M)(implicit codec: HttpContentCodec[M, A]): ContentCodec[A] =
```

i.e. proof that you can handle that media type. Intersection for composition (e.g. `HttpContentCodec[`application/json` with `text/html`]`).

It'd be a bit more complicated, however, because `M` could be inferred but `A` cannot be, so would need partial application trick to ease type inference.

Reply:
Currently the media type is also not reflected in the type of the val. So I would need to change the whole media type impl and make it look similar to `Header` where each heder impl. has its own type.

And I am understanding, that you could see the media type in the type signature, but I don't see how you could utilize it besides a human reading it. And from my experience, I don't have type annotations on my endpoint vals.

Maybe if one would explicitly define the content/accept header we could check that  you can only use valid MediaTypes. But then these headers need to be parameterized too.

Sounds like a) a lot of work and maybe too much for this pr? and b) like something that is not obviously a great change.

Reply:
Yeah, I think I would skip it. It just occurred to me it was possible.